src/backend/storage/buffer/localbuf.c

```diff
@@ -514,7 +539,10 @@ InitLocalBuffers(void)
 	LocalBufferDescriptors = (BufferDesc *) calloc(nbufs, sizeof(BufferDesc));
 	LocalBufferBlockPointers = (Block *) calloc(nbufs, sizeof(Block));
 	LocalRefCount = (int32 *) calloc(nbufs, sizeof(int32));
-	if (!LocalBufferDescriptors || !LocalBufferBlockPointers || !LocalRefCount)
+	/* Remotexact */
+	LocalBufferRemoteDescriptors = (RemoteBufferDesc *) calloc(nbufs, sizeof(RemoteBufferDesc));
```

Comment:
Should we increase nbufs now that we won't be writing some local bufs to disk?

Reply:
One nice thing about local buffer is that you can increase the number of buffers without much overhead, and only the transaction that need the extra buffers would allocate memory for the pages. 

> A session will allocate temporary buffers as needed up to the limit given by temp_buffers. The cost of setting a large value in sessions that do not actually need many temporary buffers is only a buffer descriptor, or about 64 bytes, per increment in temp_buffers. However if a buffer is actually used an additional 8192 bytes will be consumed for it (or in general, BLCKSZ bytes).

see `temp_buffers` in https://www.postgresql.org/docs/current/runtime-config-resource.html 

So, I think we can just crank the number of buffers up for now while reducing the size of the batch when loading data. I'll also put a warning (or error?) to make it clear that we are throwing a modified page if that ever happens.

Reply:
Actually, I can just skip the remote pages in the clock sweep run, and the `no empty local buffer available` error will be triggered when we don't have enough buffers.

Reply:
Updated